### ISSUE TITLE
object detection piéce metalique with camera basler

### ISSUE THREAD

**Issue by ninja8756** [ultralytics/yolov5#13521]
### Search before asking

- [x] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hi ✌ I am working on object Detection with yolov5 and I want to use basler camera to detect objects. how can I do that and get frames from the Basler camera and make detection?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @ninja8756, thank you for your interest in YOLOv5 🚀! Please visit our ⭐️ [Tutorials](https://docs.ultralytics.com/yolov5/) to get started, where you can find quickstart guides for simple tasks like [Custom Data Training](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data/) all the way to advanced concepts like [Hyperparameter Evolution](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution/).

If this is a ❓ Question about using the Basler camera with YOLOv5, please ensure you have the necessary software to interface with your camera (e.g., Basler's [pypylon](https://github.com/basler/pypylon) library for Python) and integrate it to capture frames for processing. If you encounter issues during integration, try sharing some additional details like setup code snippets, errors, or steps causing difficulties, so we can better assist you.  

If you believe this is a 🐛 Bug Report, please provide a **minimum reproducible example** (MRE), including code and steps to reproduce the issue, as well as details on your environment setup.

## Requirements

[**Python>=3.8.0**](https://www.python.org/) with all [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) installed including [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/). To get started:
```bash
git clone https://github.com/ultralytics/yolov5  # clone
cd yolov5
pip install -r requirements.txt  # install
```

## Environments

YOLOv5 can be run in a variety of environments for seamless experimentation and deployment, including:

- **Notebooks** with free GPU: <a href="https://bit.ly/yolov5-paperspace-notebook"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml"><img src="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml/badge.svg" alt="YOLOv5 CI"></a>

If this badge is green, all [YOLOv5 GitHub Actions](https://github.com/ultralytics/yolov5/actions) Continuous Integration (CI) tests are currently passing. CI tests verify correct operation of YOLOv5 [training](https://github.com/ultralytics/yolov5/blob/master/train.py), [validation](https://github.com/ultralytics/yolov5/blob/master/val.py), [inference](https://github.com/ultralytics/yolov5/blob/master/detect.py), [export](https://github.com/ultralytics/yolov5/blob/master/export.py) and [benchmarks](https://github.com/ultralytics/yolov5/blob/master/benchmarks.py) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Please note this is an automated response. An Ultralytics engineer will review your question and provide further assistance as soon as possible. Thank you for your patience! 😊

**Comment by pderrenger** [comment]
@ninja8756 to integrate a Basler camera with YOLOv5 for object detection, you can use the `pypylon` library to capture frames and pass them to YOLOv5. Here's a minimal example:

```python
from ultralytics import YOLO
from pypylon import pylon

model = YOLO("yolov5s.pt")  # Load YOLOv5 model
camera = pylon.InstantCamera(pylon.TlFactory.GetInstance().CreateFirstDevice())
camera.StartGrabbing(pylon.GrabStrategy_LatestImageOnly)

while camera.IsGrabbing():
    grabResult = camera.RetrieveResult(5000, pylon.TimeoutHandling_ThrowException)
    if grabResult.GrabSucceeded():
        img = grabResult.Array  # Get numpy array
        results = model.predict(img)  # YOLOv5 inference
        results[0].show()
```

Install pypylon: `pip install pypylon`. For more details, see the Ultralytics YOLOv5 Predict documentation. Adjust camera settings as needed via Basler's Pylon API.
